library/src/google-maps-api-loader.ts

```diff
@@ -0,0 +1,213 @@
+      // reject the promise. This is typically the case in react dev mode when
+      // load/unload are called from a hook.
+      if (this.loadingState !== LoadingState.QUEUED) {
+        console.debug('api-loader: no longer queued');
+        reject(new Error('map loading canceled'));
+        return;
+      }
+
+      console.debug('api-loader: start actually loading');
+      this.loadingState = LoadingState.LOADING;
+      const url = this.getApiUrl(this.params);
+
+      // setup the callback
+      window.__gmcb__ = () => {
+        console.debug(`api-loader: callback called, loading complete`);
+        this.loadingState = LoadingState.LOADED;
+        resolve();
+      };
+      url.searchParams.set('callback', '__gmcb__');
+      console.debug(`api-loader: URL: ${url}`);
+
+      // create the script
+      const script = document.createElement('script');
+      script.src = url.toString();
+      script.onerror = err => reject(err);
+
+      document.head.appendChild(script);
+    });
+
+    return this.loadPromise;
+  }
+
+  /**
+   * Unloads the Google Maps API by canceling any pending script downloads
+   * and removing the global `google.maps` object.
+   */
+  static unload(): void {
+    void this.unloadAsync();
+  }
+
+  private static async unloadAsync(): Promise<void> {
+    // if loading hasn't started, reset the loadingState.
+    // this will cause the loading to not start
+    if (this.loadingState <= LoadingState.QUEUED) {
+      this.loadingState = LoadingState.UNLOADED;
+      return;
+    }
+
+    const gmScriptTags = Array.from(
+      document.querySelectorAll(`script[src^="${MAPS_API_BASE_URL}"]`)
+    );
+    this.loadingState = LoadingState.UNLOADING;
+
+    // defer to the microtask-queue and check if the loading-state was
+    // changed in the meantime. If that is the case, unload was likely called
+    // in error (or by the React dev-mode calling the effect cleanup function
+    // just for fun). In this case, it is just ignored.
+    await Promise.resolve();
+
+    if (this.loadingState !== LoadingState.UNLOADING) {
+      return;
+    }
+
+    // The elements are removed from the document and adopted into a different
+    // one. This prevents the script from executing once it's loaded if it hasn't
+    // already.
+    for (const el of gmScriptTags) {
+      el.remove();
+      tmpDoc.adoptNode(el);
+    }
+
+    if (window.google && window.google.maps) {
+      // @ts-ignore
+      delete window.google.maps;
+    }
+  }
+
+  private static getApiUrl(params: ApiParams | null): URL {
+    if (params === null) {
+      throw new Error("api-params can't be null");
+    }
+
+    const url = new URL(MAPS_API_BASE_URL);
+    for (const [key, value] of Object.entries(params)) {
+      if (value === undefined) {
+        continue;
+      }
+
+      url.searchParams.set(
+        // camelCase to snake_case conversion
+        key.replace(/[A-Z]/g, c => `_${c.toLowerCase()}`),
```

Comment:
I get what this is doing, and it is a good fix (probably didn't work before right?). But as it is only one parameter that needs to be converted into snake_case and because of better readability I would suggest to only check for this one parameter and adjust it.
```

    for (const [key, value] of Object.entries(params)) {
      if (value === undefined) {
        continue;
      }

      if (key === 'authReferrerPolicy') {
        url.searchParams.set('auth_referrer_policy', value);
      }

      url.searchParams.set(key, value);
    }
```